src/io/flutter/run/daemon/DeviceDaemon.java

```diff
@@ -148,13 +151,14 @@ static Command chooseCommand(@NotNull final Project project) {
   }
 
   private static boolean usesFlutter(@NotNull final Project project) {
-    final Workspace workspace = WorkspaceCache.getInstance(project).getNow();
-    if (workspace != null) {
-      return workspace.usesFlutter(project);
-    }
-    else {
-      return FlutterModuleUtils.hasFlutterModule(project);
+    final WorkspaceCache workspaceCache = WorkspaceCache.getInstance(project);
```

Comment:
Hmmmm.  I think I'd rather see an NPE here since it means that the ServiceManager isn't finding a registration for `WorkspaceCache.class`.  Alternatively another error since this signals we're not in a good state.